src/web/mjs/engine/DiscordPresence.mjs

```diff
@@ -0,0 +1,167 @@
+const discordPresenceId = '726702836775256094';
+const DiscordRPC = require('discord-rpc');
+DiscordRPC.register(discordPresenceId);
+
+export default class DiscordPresence {
+
+    constructor(settings) {
+        this.rpc = null;
+        this.IpcBytes = 0; // keep track of IPC connection
+
+        this._settings = settings; // Engine.Settings
+        this.enabled = false;
+        this.enabledHentai = false;
+        this.hentai = false; // Is current item hentai?
+
+        this._settings.addEventListener('loaded', this._onSettingsChanged.bind(this));
+        this._settings.addEventListener('saved', this._onSettingsChanged.bind(this));
+
+        // Current status
+        this.status = {
+            largeImageKey: 'logo',
+            largeImageText: 'Manga & Anime Downloader for Linux, Windows & MacOS'
+        };
+        this.statusNew = true;
+
+        // EvetnListener for:
+        // Connector selected
+        document.addEventListener( EventListener.onSelectConnector, this._onSelectConnector.bind(this) );
+        // Manga selected
+        document.addEventListener( EventListener.onMangaClicked, this._onMangaClicked.bind(this) );
+        // get chapter pages (Reader/Downloader)
+        document.addEventListener( EventListener.onChapterGetPages, this._onChapterGetPages.bind(this) );
+
+        this.startDiscordPresence();
+    }
+
+    _onSettingsChanged() {
+        switch (this._settings.discordPresence.value) {
+            case 'none':
+                this.enabled = false;
+                this.enabledHentai = false;
+                break;
+            case 'nohentai':
+                this.enabled = true;
+                this.enabledHentai = false;
+                break;
+            case 'hentai':
+                this.enabled = true;
+                this.enabledHentai = true;
+                break;
+            default:
+                this.enabled = false;
+                this.enabledHentai = false;
+        }
+
+        if (this.enabled) {
+            this.statusNew = true;
+            this.startDiscordPresence();
+        } else {
+            this.stopDiscordPresence();
+        }
+    }
+
+    _onSelectConnector(e) {
+        // Hentai check
+        if(e.detail.tags.includes('hentai') || e.detail.tags.includes('porn')) {
```

Comment:
Not sure if it is reliable to use the website tags.
From technical point the comparison is case sensitive which will fail for `Hentai`.
There might also be upcoming tags like `Yuri` which will very likely being forgotten to add here...
A website may not be tagged as `hentai`, but still contain some NSFW mangas.

A show all or nothing approach is probably easier to handle from a developer point of view

Reply:
What else to use :) That's what tags are for. Changed comparison to `e.detail.tags.map(t => t.toLowerCase()).includes('hentai')` and `e.detail.tags.map(t => t.toLowerCase()).includes('porn')`

Reply:
Regex might be more convenient for case insensitive (partial) match
Full with Single: `/^sample$/i.test('SaMpLe')`
Partial with Multi: `/(hentai|porn)/i.test(tag)`